### DISCUSSION TITLE
Distance vs. RSSI, Time to right distance, App for iwatch...

### DISCUSSION THREAD

**Discussion by chrisgott** [mKeRix/room-assistant#774] [General]
I'm playing around with RA for a couple Days now and collected few questions I can't answer by myself...

- BT Classic: I wanna Setup the minimum strength so my devices won't be marked as in this room when im not. The only way to do so is to set a minRSSI in the config. But there is no way to find out what's the actual RSSI is. RA sends out a distance measurement instead, but no possibility to set a minDistance. I'ld like to have both; measure the RSSI and distance, and configure with minRSSI and minDistance.
- BT Classic: I get a distance sensor value every 10 seconds wich is fine, but it still needs round half a hour from 17m distance to 0.5m distance but I only need round 1 Minute to place my device from 17m to 0.5 meter. On the Data I assume there is rounding involved, but no word about that in the manual (only in BLE Section).
- BLE first doesn't work at all, cause my device changed its MAC Adress frequently and there is no way to turn that off. But there is the App wich offers a stable ID to RA with it it works fine. Sadly, the App is not for the iWatch and this would make most sense, cause the phone can be everywhere in the house, the watch is always with me.

Hopefully I didn't doing anything wrong with RA; if so i'ld love to get support for that. Can we hope for a update?
chris

**Comment by mKeRix** [comment]
Let me try to answer your questions in the order that you asked them:

- For BT Classic, the RSSI is actually just `distance * -1`. So if you have a distance of `5` reported and you want to make that your cut-off you'd configure a `minRssi` of `-5`.
- room-assistant utilizes Kalman filtering - unfortunately this seems to be misconfigured in recent versions of room-assistant, causing annoying behavior for larger homes. Since my test setup doesn't span quite as much room I didn't catch this before releasing. I opened #775 to track this issue. You can try downgrading to v2.15.0 if you don't need any of the features that were added after that version, this has the old filtering parameters. I'd be interested to hear if those work better for you.
- When I worked on the app I read a bunch of posts that the Apple Watch doesn't support the APIs that I used for the app. I just looked at it again and the docs seem to suggest that this is actually supported now - I'm still cautious, but at least I'll check it out. Issue for tracking: #776.

**Reply by chrisgott** [reply]
thx for reply; I downgraded to 2.15.0 as suggested, played around further but noticed no big difference. for more fun while playing around I added another pi in the next room, integrated mqtt instead of homeassistant and so on. my expectation was to find a note somewhere in the mqtt message saying device X is in room Y or similar, based on strongest signal on the 'stations', but this seams not being the concept of roomassistant. I need to dig in deeper. sometimes im dreaming of a one single config file for the whole thing, stations running on a esp32 in every room (with included climate sensors) and a homeassistant entity saying im in the basement, wife is in the Livingroom and all devices are set according to that information... I would build such, but too weak on coding for that.
